gemini/getting-started/intro_gemini_2_0_flash_thinking_mode.ipynb

```diff
@@ -0,0 +1,579 @@
+        "\n",
+        "Learn more about [setting up a project and a development environment](https://cloud.google.com/vertex-ai/docs/start/cloud-environment)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Nqwi-5ufWp_B"
+      },
+      "outputs": [],
+      "source": [
+        "PROJECT_ID = \"[your-project-id]\"  # @param {type: \"string\"}\n",
+        "if not PROJECT_ID or PROJECT_ID == \"[your-project-id]\":\n",
+        "    PROJECT_ID = str(os.environ.get(\"GOOGLE_CLOUD_PROJECT\"))\n",
+        "\n",
+        "LOCATION = os.environ.get(\"GOOGLE_CLOUD_REGION\", \"us-central1\")"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "T-tiytzQE0uM"
+      },
+      "outputs": [],
+      "source": [
+        "client = genai.Client(\n",
+        "    vertexai=True,\n",
+        "    project=PROJECT_ID,\n",
+        "    location=LOCATION,\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "w0u6hYSleE0H"
+      },
+      "source": [
+        "## Use the Gemini 2.0 Flash Thinking Mode\n",
+        "\n",
+        "The following examples are some complex tasks of what the Gemini 2.0 Flash Thinking mode can solve. In each of examples you can try using different models to see how this new model compares to other models. In some cases, you'll still get the good answer from the other models, in that case, re-run it a couple of times and you'll see that Gemini 2.0 Thinking mode is more consistent thanks to its thinking step."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5M7EKckIYVFy"
+      },
+      "source": [
+        "### Set model ID\n",
+        "\n",
+        "See the [Google models](https://cloud.google.com/vertex-ai/generative-ai/docs/learn/models) page for more information."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "-coEslfWPrxo"
+      },
+      "outputs": [],
+      "source": [
+        "MODEL_ID = \"gemini-2.0-flash-thinking-exp-1219\"  # @param {type: \"string\"}"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "IrRBg9UGC9nn"
+      },
+      "source": [
+        "### **Example 1**: Code simplification\n",
+        "\n",
+        "First, try with a simple code comprehension and simplification example."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "dLhhffx2C9nn"
+      },
+      "outputs": [],
+      "source": [
+        "response = client.models.generate_content(\n",
+        "    model=MODEL_ID,\n",
+        "    contents=\"How can I simplify this? `(Math.round(radius/pixelsPerMile * 10) / 10).toFixed(1);`\",\n",
```

Comment:
Can you bring this prompt out into a variable with multiline strings to make it easier to read?